src/components/org/apache/jmeter/config/CSVDataSet.java

```diff
@@ -146,71 +146,76 @@ public void setProperty(JMeterProperty property) {
 
     @Override
     public void iterationStart(LoopIterationEvent iterEvent) {
-        FileServer server = FileServer.getFileServer();
-        final JMeterContext context = getThreadContext();
-        String delim = getDelimiter();
-        if (delim.equals("\\t")) { // $NON-NLS-1$
-            delim = "\t";// Make it easier to enter a Tab // $NON-NLS-1$
-        } else if (delim.length()==0){
-            log.warn("Empty delimiter converted to ','");
-            delim=",";
-        }
-        if (vars == null) {
-            String _fileName = getFilename();
-            String mode = getShareMode();
-            int modeInt = CSVDataSetBeanInfo.getShareModeAsInt(mode);
-            switch(modeInt){
-                case CSVDataSetBeanInfo.SHARE_ALL:
-                    alias = _fileName;
-                    break;
-                case CSVDataSetBeanInfo.SHARE_GROUP:
-                    alias = _fileName+"@"+System.identityHashCode(context.getThreadGroup());
-                    break;
-                case CSVDataSetBeanInfo.SHARE_THREAD:
-                    alias = _fileName+"@"+System.identityHashCode(context.getThread());
-                    break;
-                default:
-                    alias = _fileName+"@"+mode; // user-specified key
-                    break;
+        String _fileName = getFilename();
+        if (!_fileName.isEmpty()) {
+            FileServer server = FileServer.getFileServer();
+            final JMeterContext context = getThreadContext();
+            String delim = getDelimiter();
+            if (delim.equals("\\t")) { // $NON-NLS-1$
+                delim = "\t";// Make it easier to enter a Tab // $NON-NLS-1$
+            } else if (delim.length()==0){
+                log.warn("Empty delimiter converted to ','");
+                delim=",";
             }
-            final String names = getVariableNames();
-            if (names == null || names.length()==0) {
-                String header = server.reserveFile(_fileName, getFileEncoding(), alias, true);
-                try {
-                    vars = CSVSaveService.csvSplitString(header, delim.charAt(0));
-                    firstLineIsNames = true;
-                } catch (IOException e) {
-                    log.warn("Could not split CSV header line",e);
+            if (vars == null) {
+                String mode = getShareMode();
+                int modeInt = CSVDataSetBeanInfo.getShareModeAsInt(mode);
+                switch(modeInt){
+                    case CSVDataSetBeanInfo.SHARE_ALL:
+                        alias = _fileName;
+                        break;
+                    case CSVDataSetBeanInfo.SHARE_GROUP:
+                        alias = _fileName+"@"+System.identityHashCode(context.getThreadGroup());
```

Comment:
Do you assume "all the thread groups would get unique hash codes". I'm afraid that would not work.